app/test/unit/git/cherry-pick-test.ts

```diff
@@ -264,4 +271,126 @@ describe('git/cherry-pick', () => {
     }
     expect(result).toBe(null)
   })
+
+  describe('cherry picking with conflicts', () => {
+    beforeEach(async () => {
+      // In the 'git/cherry-pick' `beforeEach`, we call `createRepository` which
+      // adds a commit to the feature branch with a file called THING.md. In
+      // order to make a conflict, we will add the same file to the target
+      // branch.
+      const conflictingCommit = {
+        commitMessage: 'Conflicting Commit!',
+        entries: [
+          {
+            path: 'THING.md',
+            contents: '# HELLO WORLD! \n CREATING CONFLICT! FUN TIMES!\n',
+          },
+        ],
+      }
+      await makeCommit(repository, conflictingCommit)
+    })
+
+    it('successfully detect cherry pick with conflicts', async () => {
+      result = await cherryPick(repository, featureBranch.tip.sha)
+      expect(result).toBe(CherryPickResult.ConflictsEncountered)
+
+      const status = await getStatusOrThrow(repository)
+      const conflictedFiles = status.workingDirectory.files.filter(
+        f => f.status.kind === AppFileStatusKind.Conflicted
+      )
+      expect(conflictedFiles).toHaveLength(1)
+    })
+
+    it('successfully continue cherry pick with conflicts after resolving them', async () => {
+      result = await cherryPick(repository, featureBranch.tip.sha)
+      expect(result).toBe(CherryPickResult.ConflictsEncountered)
+
+      const statusAfterCherryPick = await getStatusOrThrow(repository)
+      const { files } = statusAfterCherryPick.workingDirectory
+
+      const diffCheckBefore = await GitProcess.exec(
+        ['diff', '--check'],
+        repository.path
+      )
+
+      expect(diffCheckBefore.exitCode).toBeGreaterThan(0)
+
+      // resolve conflicts by writing files to disk
+      await FSE.writeFile(
+        Path.join(repository.path, 'THING.md'),
+        '# HELLO WORLD! \nTHINGS GO HERE\nFEATURE BRANCH UNDERWAY\n'
+      )
+
+      const diffCheckAfter = await GitProcess.exec(
+        ['diff', '--check'],
+        repository.path
+      )
+
+      expect(diffCheckAfter.exitCode).toEqual(0)
```

Comment:
Just curious: what is this `git diff --check` for? Should we add a comment to explain _why_ this was added? 🤔

Reply:
It is just a way to check if conflicts exist. It was being used in rebase testing for conflicts so I just followed the practice. (After having the same question as you tho so good point adding documentation of it) https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---check